Documentation/design-docs/host-components.md

```diff
@@ -2,32 +2,33 @@
 
 The .NET Core default hosting setup consists of several components which are described here.
 
-## Executable
-The executable, which is also sometimes refered to as the "host" can be either the
-* `dotnet.exe` - which comes from a shared location and is typically the latest version available on the machine
-* `apphost.exe` - which is used in case of self-contained apps. The name of the .exe will be app dependent and it comes with the app
-
-The executable does just one thing, it finds the `hostfxr.dll` (for Windows, Linux and Max use their respective naming schemes) and passes control to it.
-For `dotnet.exe` host the hostfxr is obtained from the `./shared/host/fxr<highestversion>` folder.
-For self-contained apps (apphost scenario):
-1. The app's folder is searched first. This is either the folder where the `apphost.exe` lives or the path it has embeded in it as the app path.
-2. If the `DOTNET_ROOT` environment variable is defined, that path is searched
-3. The default shared locations are searched
+## Entry-point hosts
+.NET Core comes with several executables/libraries which act as the main entry-point to start code execution. These are typically referred to as the "host":
+* `dotnet` (executable) - which comes from a shared location and is typically the latest version available on the machine. This is also sometimes called the "muxer".
+* `apphost` (executable) - which is used to give app an actual executable which can be run directly. The executable will be named using the application name. The advantage of having an app-local executable is that it can be customized for each app (not just the name, but icon, OS behavior and so on).
+* `comhost` (library) - which is used to enable COM server hosting. Component which wants to expose COM server objects will be built with this dynamic library in its output. The `comhost` then acts as the main entry point for the OS.
+
+The executable does just one thing, it finds the `hostfxr` library and passes control to it. It also exposes the right entry points for its purpose (so the "main" for `dotnet` and `apphost`, the COM exports for `comhost` and so on).
+* `dotnet` host - `hostfxr` is obtained from the `./shared/host/fxr<highestversion>` folder (relative to the location of the `dotnet` host).
+* `apphost` and `comhost` - `hostfxr` is located by
+    1. The app's folder is searched first. This is either the folder where the `apphost` or `comhost` lives or in case of `apphost` it is the path it has embedded in it as the app path.
+    1. If the `DOTNET_ROOT` environment variable is defined, that path is searched
+    1. The default shared locations are searched
```

Comment:
do we want to elaborate on the default shared locations?  Registry... Do we want this to be forward looking?  i.e Linux /etc/dotnet/config & $HOME/.dotnet/config?

Reply:
I would rather keep this as is for now. I actually thought about specifically this and the registry, but for now I would leave it out once we have the story fully figured out.